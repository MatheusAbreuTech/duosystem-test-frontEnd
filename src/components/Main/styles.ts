'use client'

import styled from 'styled-components'

export const Wrapper = styled.main`
  max-width: 80rem;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 2rem;
  margin: 0 auto;
`

export const Title = styled.h1`
  font-size: 2.5rem;
`
