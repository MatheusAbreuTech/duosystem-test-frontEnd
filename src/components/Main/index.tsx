import * as S from './styles'
import TaskList from '../TaskList'
import AddTask from '../AddTask'
import TaskSearch from '../TaskSearch'

const Main = () => {
  return (
    <S.Wrapper>
      <S.Title>Lista de tarefas</S.Title>

      {/* input de pesquisa */}
      <TaskSearch />

      {/* listagem */}
      <TaskList />

      {/* botao para adicionar umas task */}
      <AddTask />
    </S.Wrapper>
  )
}

export default Main
