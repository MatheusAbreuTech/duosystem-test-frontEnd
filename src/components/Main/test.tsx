import { render } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the Main component in screen', () => {
    // renderiza o component
    const { container } = render(<Main />)

    expect(container).toBeInTheDocument()
  })
})
