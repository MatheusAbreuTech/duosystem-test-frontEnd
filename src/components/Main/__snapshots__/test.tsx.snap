// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the Main component in screen 1`] = `
.c0 {
  max-width: 80rem;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 2rem;
  margin: 0 auto;
}

.c1 {
  font-size: 2.5rem;
}

.c7 {
  background-color: transparent;
  color: #0c2e73;
  border: none;
  border-radius: 0.5rem;
  padding: 1rem;
  max-height: 4.2rem;
  height: 100%;
  cursor: pointer;
  font-size: 1.6rem;
}

.c10 {
  background-color: #0c2e73;
  color: #ffffff;
  border: none;
  border-radius: 0.5rem;
  padding: 1rem;
  max-height: 4.2rem;
  height: 100%;
  cursor: pointer;
  font-size: 1.6rem;
}

.c5 {
  width: 100%;
}

.c6 {
  max-width: 70%;
  height: auto;
  display: flex;
  align-items: center;
  gap: 1rem;
}

.c6 >span {
  font-size: 1.6rem;
}

.c3 {
  width: 100%;
  height: 42px;
  padding: 0 1rem;
  background-color: #fff;
  outline: none;
  border: none;
  font-size: 1.6rem;
  border-radius: 5px;
}

.c8 {
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  width: 100%;
  align-items: flex-start;
  gap: 0.5rem;
}

.c8 >label {
  font-size: 1.6rem;
}

.c9 {
  display: flex;
  align-items: center;
  gap: 1rem;
  width: 100%;
}

.c9 >input {
  border: 1px solid #383838;
}

.c2 {
  width: 100%;
  height: 44px;
  padding-right: 1rem;
  border-radius: 5px;
  background-color: #fff;
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px solid #383838;
}

.c4 {
  border: none;
  outline: none;
  background-color: transparent;
  cursor: pointer;
}

@media (max-width: 700px) {
  .c6 {
    max-width: 100%;
  }
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Lista de tarefas
  </h1>
  <form
    action=""
    class="c2"
  >
    <input
      class="c3"
      data-testid="searchValue"
      name="searchValue"
    />
    <button
      class="c4"
      type="submit"
    >
      <img
        alt=""
        height="20"
        src="/img/search.svg"
        width="20"
      />
    </button>
  </form>
  <div
    class="c5"
  >
    <div
      class="c6"
    >
      <span>
        Filtrar por: 
      </span>
      <button
        class="c7"
        color="transparent"
        data-testid="BtnFilterByAll"
      >
        Todas
      </button>
      <button
        class="c7"
        color="transparent"
        data-testid="BtnFilterByStatus"
      >
        Concluidas
      </button>
      <button
        class="c7"
        color="transparent"
        data-testid="BtnFilterCreatedDate"
      >
        Data de criação
      </button>
    </div>
    <p>
      Nenhuma tarefa encontrada
    </p>
  </div>
  <form
    action=""
    class="c8"
  >
    <label
      for="taskDescription"
    >
      Adicionar tarefa:
    </label>
    <div
      class="c9"
    >
      <input
        class="c3"
        data-testid="taskDescription"
        id="taskDescription"
        name="taskDescription"
        type="text"
      />
      <button
        class="c10"
        color="darkBlue"
        type="submit"
      >
        Adicionar
      </button>
    </div>
  </form>
</main>
`;
